website/cue/reference/remap/functions/get_enrichment_table_record.cue

```diff
@@ -0,0 +1,80 @@
+package metadata
+
+remap: functions: get_enrichment_table_record: {
+	category:    "Enrichment"
+	description: """
+		Searches an [enrichment table](\(urls.enrichment_tables_concept)) for a row that matches the
+		provided condition. A single row must be matched. If either no rows or more than one row is
+		found, an error is returned.
+
+		\(remap._enrichment_table_explainer)
+		"""
+
+	arguments: [
+		{
+			name:        "table"
+			description: "The [enrichment table](\(urls.enrichment_tables_concept)) to search."
+			required:    true
+			type: ["string"]
+		},
+		{
+			name: "condition"
+			description: """
+				The condition to search on. Since the condition is used at boot time to create
+				indices into the data, these conditions must be statically defined.
+				"""
+			required: true
+			type: ["object"]
+		},
+		{
+			name: "select"
+			description: """
+				A subset of fields from the enrichment table to return. If not specified,
+				all fields are returned.
+				"""
+			required: false
+			type: ["array"]
+		},
+		{
+			name:        "case_sensitive"
+			description: "Should text fields match case exactly."
+			required:    false
+			type: ["boolean"]
+			default: true
+		},
+	]
+	internal_failure_reasons: [
+		"The row isn't found",
+		"Multiple rows are found that match the condition",
+	]
```

Comment:
I suspect people will want to differentiate between these two errors. There's no actionable here right now, but I do wonder if at some point we'll want to switch to typed errors in VRL, so that people can pattern-match against them.

Reply:
Yeah, this came up before too (I think in the PR adding the functions? Or maybe the RFC?). I completely agree that users are likely to want to differentiate these errors. I think we will probably have more VRL functions in the future that have discrete errors too. Definitely worth mulling about.